engines/osxnative/engine.go

```diff
@@ -49,6 +67,33 @@ var payloadSchema = schematypes.Object{
 	Required: []string{"command"},
 }
 
+var configSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"createUser": schematypes.Boolean{
+			MetaData: schematypes.MetaData{
+				Title: "Tells if a user should be created to run a command",
+				Description: `When set to true, a new user is created on the fly to run
+				the command. It runs the command from whitin the user's home directory.`,
+			},
+		},
+		"userGroups": schematypes.Array{
+			MetaData: schematypes.MetaData{
+				Title: "List of groups to assign user to",
+				Description: `It contains the list of groups the user will belong to.
+				The first item is the user's primary group, the rest will be user's
+				supplementary groups.`,
+			},
+			Items:  schematypes.String{},
+			Unique: true,
+		},
+	},
+	Required: []string{"createUser"},
+}
+
+func (e engineProvider) ConfigSchema() schematypes.Schema {
+	return configSchema
+}
+
 func (e engine) PayloadSchema() schematypes.Object {
```

Comment:
`PayloadSchema` expects it by value in `EngineBase`. If I use a pointer here, there will be a conflict in the method signature. I had problems with this in the past. We can address it in another PR.